tortoise/state.go

```diff
@@ -134,25 +137,31 @@ func (s *state) Recover() error {
 	it = s.db.Find([]byte(namespaceOpinions))
 	defer it.Release()
 	for it.Next() {
-		layer := decodeLayerKey(it.Key())
-		offset := 1 + types.LayerIDSize
-		block1 := decodeBlock(it.Key()[offset : offset+types.BlockIDSize])
+		// first byte is namespace
+		layer1 := decodeLayerKey(it.Key()[1:])
+		offset1 := 1 + types.LayerIDSize
+		offset2 := offset1 + types.BlockIDSize + types.LayerIDSize
```

Comment:
i think it's more readable to do
layer1Offset := len(namespaceOpinions)
block1Offset := layer1Offset + types.LayerIDSize
layer2Offset := block1OffSet + types.BlockIDSize
block2Offset := layer2Offset + types.LayerIDSize